### PR TITLE
Support azure migrate assessments

### LINKED ISSUE #425: dynamic resource methods with float values are being ignored
# Describe the problem
if properties attribute for any azure resource has float values than that attribute is being ignored while creating instance methods 

Executing the below in context of an Azure resource, 
```
response =  { properties: { scalingFactor: 1.0 } }

resource.create_resource_methods(response)
```

expected: 
```
resource.properties.scalingFactor => 1.0 
```
actual:
```
resource.properties.scalingFactor => NullResponse
```

## Possible Solution
allow `create_resource_methods` to support `Float` and `Bignum` types 
long term support is to include all `Numeric` types 

snippet with issue
```
case value.class.to_s
    when 'String', 'Integer', 'TrueClass', 'FalseClass', 'Fixnum', 'Time'
```

temporary solution: 
```
case value.class.to_s
    when 'String', 'Integer', 'TrueClass', 'FalseClass', 'Fixnum', 'Time', 'Bignum', 'Float'
```

Long term solution:

```
case value
    when String, Numeric, TrueClass, FalseClass, Time
```

### PR DESCRIPTION
**Pull request by sathish-progress**
### Description
Support singular and plural resource for Azure Migrate Assessment(s)

Also fixed #425 as part of this

### Issues Resolved
#425 

### Check List

- [X] New functionality includes tests
- [X] All tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by sa-progress — COMMENTED** [review]
azureStorageRedundancy has other 4 types : GeoRedundant, ReadAccessGeoRedundant, Unknown, ZoneRedundant.
Shouldn't we provide this value as default value in variable.tf and other values like scalingFactor as well?

**Comment by sathish-progress** [comment]
> azureStorageRedundancy has other 4 types : GeoRedundant, ReadAccessGeoRedundant, Unknown, ZoneRedundant.
> Shouldn't we provide this value as default value in variable.tf and other values like scalingFactor as well?

sorry I didn't understand, this PR is about Azure Migrate Assessments and not about Azure Storage Redundancy ..

File: docs/resources/azure_migrate_assessment.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+```ruby
+describe azure_migrate_assessment(resource_group: 'migrated_vms', project_name: 'zoneA_migrate_assessment_project', group_name: 'zoneA_machines_group', name: 'zoneA_machines_migrate_assessment') do
+  it  { should exist }
+end
+```
+## Parameters
+
+| Name           | Description                                                                      |
+|----------------|----------------------------------------------------------------------------------|
+| name           | Name of the Azure Migrate Assessments to test.                                   |
```
Comment (sa-progress) [inline docs/resources/azure_migrate_assessment.md:48]: Name is twice .Please check once .

File: libraries/azure_migrate_assessment.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  def initialize(opts = {})
+    raise ArgumentError, 'Parameters must be provided in an Hash object.' unless opts.is_a?(Hash)
+
+    opts[:resource_provider] = specific_resource_constraint('Microsoft.Migrate/assessmentProjects', opts)
+    opts[:required_parameters] = %i(project_name group_name name)
+    opts[:resource_path] = [opts[:project_name], 'groups', opts[:group_name], 'assessments'].join('/')
+    super(opts, true)
+  end
+end
```
Comment (sa-progress) [inline libraries/azure_migrate_assessment.rb:20]: to_s implementation is missing .

File: libraries/azure_migrate_assessments.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  private
+
+  def populate_table
+    @resources.each do |resource|
+      @table << resource.merge(resource[:properties])
+    end
+  end
+end
```
Comment (sa-progress) [inline libraries/azure_migrate_assessments.rb:31]: to_s implementation is missing .

**Review by sa-progress — COMMENTED** [review]
There are minor comments please check.

File: libraries/azure_migrate_assessment.rb (hunk 2 above)
Comment (sathish-progress) [reply libraries/azure_migrate_assessment.rb:24]: done

File: libraries/azure_migrate_assessments.rb (hunk 3 above)
Comment (sathish-progress) [reply libraries/azure_migrate_assessments.rb:34]: done

**Review by sa-progress — APPROVED** [review]
LGTM

**Review by IanMadd — COMMENTED** [review]
Feedback for @dkumaras

File: docs/resources/azure_migrate_assessment.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+## Availability
+
+### Installation
+
+This resource is available in the [InSpec Azure resource pack](https://github.com/inspec/inspec-azure). For an example, `inspec.yml` file and how to set up your Azure credentials, refer to resource pack [README](../../README.md#Service-Principal).
+
+## Syntax
+
+`name` is a required parameter, and `resource_group` is an optional parameter.
```
[the remark this answers is not in the document]
Comment (sathish-progress) [reply docs/resources/azure_migrate_assessment.md]: this is done

**Comment by sathish-progress** [comment]
@dkumaras , Could you please amend your signature to your commits that is causing the DCO to fail?

**Review by IanMadd — APPROVED** [review]

**Pull request merged**